KenticoOnboardingApplication/src/KenticoOnboardingApplication.Services/Services/UpdateItemService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading.Tasks;
+using KenticoOnboardingApplication.Contracts.Helpers;
+using KenticoOnboardingApplication.Contracts.Models;
+using KenticoOnboardingApplication.Contracts.Repositories;
+using KenticoOnboardingApplication.Contracts.Services;
+using KenticoOnboardingApplication.Contracts.Services.Wrappers;
+
+namespace KenticoOnboardingApplication.Services.Services
+{
+    internal class UpdateItemService : IUpdateItemService
+    {
+        private readonly IListRepository _repository;
+        private readonly ITimeManager _timeManager;
+        private readonly IGetItemService _getItemService;
+
+        public UpdateItemService(IListRepository repository, ITimeManager timeManager, IGetItemService getItemService)
+        {
+            _repository = repository;
+            _timeManager = timeManager;
+            _getItemService = getItemService;
+        }
+
+        public async Task<RetrievedItem<Item>> UpdateItemAsync(Item item)
+        {
+            var retrievedItem = await _getItemService.GetItemAsync(item.Id);
+            if (!retrievedItem.WasFound)
+            {
+                return retrievedItem;
+            }
+
+            var updatedItem = CreateUpdatedItem(existingItem: retrievedItem.Item, newItem: item);
```

Comment:
[👎 ](https://trolls.gg/wp-content/uploads/2018/01/not-good-enough.jpg)
```suggestion
            var updatedItem = CreateUpdatedItem(retrievedItem, item);
```
(while naming parameters definitely works, putting a type-guard in charge is much more effective 😉)